hazelcast/src/hazelcast/client/sql.cpp

```diff
@@ -193,6 +252,21 @@ sql_service::handle_execute_response(
           response.error->code,
           std::move(response.error->message),
           std::move(response.error->suggestion)));
+    } else {
+        if (is_smart_routing_ && response.partition_argument_index !=
+                                   original_partition_argument_index) {
+            if (response.partition_argument_index != -1) {
+                partition_argument_index_cache_->put(
+                  sql_query,
+                  std::make_shared<int32_t>(response.partition_argument_index));
+                auto temp_shared_ptr = statement_par_arg_index_ptr.lock();
+                if (temp_shared_ptr) {
+                    temp_shared_ptr->store(response.partition_argument_index);
+                }
```

Comment:
```suggestion
                if (auto argument_index = statement_par_arg_index_ptr.lock()) {
                    argument_index->store(response.partition_argument_index);
                }
```

Reply:
Can you rename the variable as well because `temp_shared_ptr` doesn't express much ?

Reply:
renamed